apps/events/migrations/0014_move_schedule_messages_off_periodic_tasks.py

```diff
@@ -0,0 +1,120 @@
+# Generated by Django 4.2.11 on 2024-07-12 14:50
+from datetime import datetime, timedelta
+from django.db import migrations
+import json
+
+from apps.events.models import EventAction, EventActionType, ScheduledMessage
+from apps.experiments.models import ExperimentSession
+
+def is_complete(task):
+	return datetime.now() > task.expires
+
+def calculate_next_run(task, current_time):
+	delta = timedelta(seconds=task.interval.every.total_seconds())
+	next_time = task.start_time
+	while next_time <= current_time:
+		next_time += delta
+	return next_time
+
+def get_interval_duration(interval):
+    if not interval:
+        return None
+    if interval.period == 'seconds':
+        return timedelta(seconds=interval.every)
+    elif interval.period == 'minutes':
+        return timedelta(minutes=interval.every)
+    elif interval.period == 'hours':
+        return timedelta(hours=interval.every)
+    elif interval.period == 'days':
+        return timedelta(days=interval.every)
+    elif interval.period == 'weeks':
+        return timedelta(weeks=interval.every)
+    else:
+        return None
+
+def calculate_remaining_repetitions(task, current_time):
+    interval_duration = get_interval_duration(task.interval)
+    remaining_time = task.expires - current_time if task.expires else None
```

Comment:
If the task never expires doesn't that imply an infinite number of repetitions? I don't think we support that but maybe we should?

Reply:
ah I didn't think about that but that makes sense. In that case how should remaining time be represented?